sunpy/map/mapbase.py

```diff
@@ -1233,9 +1233,11 @@ def coordinate_system(self):
         # changes it to a ctype that is understood.  See Thompson, 2006, A.&A.,
         # 449, 791.
         if ctype1.lower() in ("solar-x", "solar_x"):
+            warn_deprecated("CTYPE1 value 'solar-x'/'solar_x' is deprecated, use 'HPLN-TAN' instead.")
```

Comment:
Can we add a generic map test which hits these warns and checks the message?

Reply:
Cool will do that